subcommands/waves/rollout.go

```diff
@@ -10,25 +15,81 @@ import (
-Upon rollout a wave becomes available as an update source for production devices in a specific
-device group.  An rollout is not instant, but rather each device in a given group will update to
-this wave's targets at some point in time which is determined by many factors: most important being
-network conditions between a device and update servers, as well as a device update schedule.`,
+	rollout := &cobra.Command{
+		Use:   "rollout <wave> -g <group>",
+		Short: "Roll out a wave to a subset of devices",
+		Long: `Roll out a wave to a subset of devices in a factory (in a device group) matching a wave's tag.
+Upon rollout a wave becomes available as an update source for production devices (in a given device group).
+A rollout is not instant, rather each device will update to the wave's targets at some point in time.
+The exact update time is determined by many factors:
+e.g. device up and down lifecycle, its update schedule, networking between a device and update servers, etc.
+At least one of command arguments is required to limit the subset of devices to roll out to.
+If you want to roll out to all matching devices in a factory, please, use the complete command.`,
 		Run:  doRolloutWave,
-		Args: cobra.ExactArgs(2),
-	})
+		Args: cobra.RangeArgs(1, 2),
+	}
+	rollout.Flags().StringP("group", "g", "", "A device group to roll out a wave to")
+	rollout.Flags().StringP("uuids", "", "",
+		`A comma-separated list of exact device UUIDs to roll out a wave to.
+Also accepts a filename containing a comma-separated list via "--uuids=@path/to/file.name".`,
+	)
+	cmd.AddCommand(rollout)
 }
 
 func doRolloutWave(cmd *cobra.Command, args []string) {
 	factory := viper.GetString("factory")
 	wave := args[0]
-	group := args[1]
-	logrus.Debugf("Rolling out a wave %s for %s to %s", wave, factory, group)
+	group := readGroup(cmd, args)
+	uuids := readUuids(cmd)
+
+	if len(group) == 0 && len(uuids) == 0 {
+		subcommands.DieNotNil(errors.New("One of flags must be set: group, uuids\n" + cmd.UsageString()))
+	}
 
-	options := client.WaveRolloutOptions{Group: group}
+	selector := "all devices"
+	if len(uuids) > 0 {
+		selector = strconv.Itoa(len(uuids)) + " devices"
+	}
+	if len(group) > 0 {
+		selector += " in " + group
+	}
+	logrus.Debugf("Rolling out a wave %s for %s to %s", wave, factory, selector)
+
+	options := client.WaveRolloutOptions{Group: group, Uuids: uuids}
 	subcommands.DieNotNil(api.FactoryRolloutWave(factory, wave, options))
 }
+
+func readGroup(cmd *cobra.Command, args []string) string {
+	// Backward-compatible reader: new way - named flag, old way - positional flag
+	group, _ := cmd.Flags().GetString("group")
+	if len(args) > 1 {
+		if len(group) > 0 {
+			subcommands.DieNotNil(errors.New(
+				"Flag \"group\" cannot be both positional and named\n" + cmd.UsageString(),
+			))
+		}
+		group = args[1]
+	}
+	return group
+}
+
+func readUuids(cmd *cobra.Command) []string {
+	uuids, _ := cmd.Flags().GetString("uuids")
```

Comment:
EdgeLock2Go in particular is slightly different than us.

Reply:
At first, my plan was to imply some size limit onto a single UUID, so that the user cannot upload a home video as _one huge uuid junk_. Probably, I need to do that.

Reply:
fixed in force-push